src/Kiota.Builder/EqualityComparers/OpenApiServerComparer.cs

```diff
@@ -1,23 +1,37 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.Diagnostics.CodeAnalysis;
-using System.Text.RegularExpressions;
 using Microsoft.OpenApi.Models;
 
 namespace Kiota.Builder.EqualityComparers;
 
-internal sealed partial class OpenApiServerComparer : IEqualityComparer<OpenApiServer>
+internal sealed class OpenApiServerComparer : IEqualityComparer<OpenApiServer>
 {
-    [GeneratedRegex("^https?://", RegexOptions.IgnoreCase, 500)]
-    private static partial Regex protocolCleanupRegex();
+    private static ReadOnlySpan<char> TrimProtocol(ReadOnlySpan<char> output)
+    {
+        output = output.TrimStart("http://");
+        output = output.TrimStart("https://");
+        return output;
```

Comment:
would this handle "HTTPS"?

Reply:
It wouldn't, but I've updated it to add this. I've also added unit tests for it.